src/software/geom/util.cpp

```diff
@@ -1251,6 +1116,172 @@ std::optional<Segment> mergeFullyOverlappingSegments(Segment segment1, Segment s
     }
 }
 
+std::optional<std::vector<Segment>> reduceParallelSegments(std::vector<Segment> segments)
```

Comment:
So what I would expect this function to do is to return the given `segments`, but with the minimum number of segments removed such that all remaining segments are parallel. Is there a reason that doing it this way instead works better? Just seems a bit more complicated.

Reply:
Oh that is not what it does at all. I'll make it more clear.

reduceParallelSegments() combines parallel Segments into the fewest possible Segments that occupy the same space as the original input

Reply:
reduceParallelSegments deleted and replace with combinoParallelSegments()